snapshots/overlay/overlay.go

```diff
@@ -419,14 +447,43 @@ func (o *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 			return fmt.Errorf("failed to create snapshot: %w", err)
 		}
 
-		if len(s.ParentIDs) > 0 {
-			st, err := os.Stat(o.upperPath(s.ParentIDs[0]))
-			if err != nil {
-				return fmt.Errorf("failed to stat parent: %w", err)
+		_, info, _, err = storage.GetInfo(ctx, key)
+		if err != nil {
+			return fmt.Errorf("failed to get snapshot info: %w", err)
+		}
+
+		mappedUID := -1
+		mappedGID := -1
+		// NOTE: if idmapped mounts' supported by hosted kernel there may be
+		// no parents at all, so overlayfs will not work and snapshotter
+		// will use bind mount. To be able to create file objects inside the
+		// rootfs -- just chown this only bound directory according to provided
+		// {uid,gid}map. In case of one/multiple parents -- chown upperdir.
+		if v, ok := info.Labels[snapshots.LabelSnapshotUIDMapping]; ok {
```

Comment:
I am thinking about whether we should check the mapping is valid at first. Right now, we just check the hostID. The container ID can be invalid. It can be handled in the follow-up.

Reply:
> I am thinking about whether we should check the mapping is valid at first. Right now, we just check the hostID. The container ID can be invalid.

You mean that container ID should only be 0? For usual correctness both mappings will be checked later in `parseIDMapping`. Here I check `hostID`, because it's going to be used right away.

Reply:
Done